src/controls/UserEditInteraction.js

```diff
@@ -0,0 +1,221 @@
+import Interaction from 'ol/interaction/Interaction.js';
+import VectorLayer from 'ol/layer/Vector.js';
+import RenderFeature, { toFeature } from 'ol/render/Feature.js';
+import VectorSource from 'ol/source/Vector.js';
+import Draw from 'ol/interaction/Draw.js';
+import Modify from 'ol/interaction/Modify.js';
+import { defaultStyles } from '../constants.js';
+import { getLayerForMapboxSourceLayer } from '../utils.js';
+
+/**
+ * @typedef {Object} Options
+ * @property {import('../YioMap.js').YioMap} yioMap
+ * @property {import('ol/Map.js').default} map ol map
+ */
+
+export default class UserSelectInteraction extends Interaction {
+  /** @type {import('../YioMap.js').YioMap} */
+  #yioMap = null;
+
+  /** @type {import('ol/Map.js').default} */
+  #map = null;
+
+  /**
+   * @type {VectorLayer} layer for temporary features of the draw / modify interactions
+   */
+  #editLayer = null;
+
+  getEditLayer() {
+    return this.#editLayer;
+  }
+
+  /**
+   * clears the source of the edit layer, containing newly drawn and modified features
+   */
+  clearEditSource() {
+    this.#editLayer.getSource().clear();
+  }
+
+  /**
+   * ol layer connected associated with the given mapbox `source-layer` (editLayer)
+   * @type {VectorLayer | import("ol/layer/VectorTile.js").default}
+   */
+  #editSourceLayer;
+
+  /**
+   * IDs of features that were modified
+   * @type {Array<string | number>}
+   */
+  #modifiedFeatureIDs = [];
+
+  /**
+   * @param {Options} options
+   */
+  constructor(options) {
+    super();
+    this.#yioMap = options.yioMap;
+    this.#map = options.map;
+
+    this.#editLayer = new VectorLayer({
+      source: new VectorSource({
+        features: [],
+      }),
+    });
+
+    this.drawInteraction = new Draw({
+      source: this.#editLayer.getSource(),
+      condition: event => {
```

Comment:
Before doing the check for existing features, you should maybe check for the default condition (I think it is `primaryAction`.